### PR TITLE
Python 3 compatibility

### LINKED ISSUE #153: Python 3 support
Mantid itself is very close to having complete support for Python 3. We should add this support to `mslice`.

This will involve:

* converting code to use constructs that both Python 2/Python 3 can understand: https://www.mantidproject.org/Python_3, using `six` where appropriate
* setting up build support to enable us to test this so that we don't introduce regressions in the future

### PR DESCRIPTION
**Pull request by MatthewBowles**
This PR aims to convert the code to be compatible with both Python 2 and 3.

**To do:**
Build support to prevent future regressions

Fixes #153

### CODE REVIEW

File: mslice/models/workspacemanager/mantid_workspace_provider.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def _processLoadedWSLimits(self, workspace):
         E2q = 2. * constants.m_n / (constants.hbar ** 2)  # Energy to (neutron momentum)^2 (==2m_n/hbar^2)
         meV2J = constants.e / 1000.                       # meV to Joules
         m2A = 1.e10                                       # metres to Angstrom
-        if ws_name not in self._limits.keys():
+        if ws_name not in list(self._limits.keys()):
```
Comment (martyngigg) [inline mslice/models/workspacemanager/mantid_workspace_provider.py]: Same with `list` and `keys()`

**Comment by MatthewBowles** [comment]
@martyngigg The new build script setup seems to be working with 2.7.

**Comment by martyngigg** [comment]
@MatthewBowles Is this ready to review again now?

**Comment by MatthewBowles** [comment]
@martyngigg Not really, the tests pass but the python 3 version still has a few bugs

**Comment by martyngigg** [comment]
@MatthewBowles Okay, thanks. Are the bugs purely in the GUI layer or is it possible to add tests to catch them?

**Comment by MatthewBowles** [comment]
@martyngigg Looks like it was just a problem with launching a new window. Should be good to go now.

File: mslice/models/workspacemanager/mantid_workspace_provider.py [hunk 2]
```diff
@@ -166,9 +168,9 @@ def get_limits(self, workspace, axis):
 
     def propagate_properties(self, old_workspace, new_workspace):
         """Propagates MSlice only properties of workspaces, e.g. limits"""
-        if old_workspace in self._EfDefined.keys():
+        if old_workspace in self._EfDefined:
             self._EfDefined[new_workspace] = self._EfDefined[old_workspace]
-        if old_workspace in self._limits.keys():
+        if old_workspace in self._limits:
```
Comment (martyngigg) [inline mslice/models/workspacemanager/mantid_workspace_provider.py]: @MatthewBowles Do you know why the `.keys` error here that was fixed in the last commit wasn't picked up by a test?

Comment (MatthewBowles) [reply mslice/models/workspacemanager/mantid_workspace_provider.py]: @martyngigg Looks like the only classes that use it are mocked out in the tests

Comment (martyngigg) [reply mslice/models/workspacemanager/mantid_workspace_provider.py]: I see. Would it be hard to write some tests for it?

Comment (MatthewBowles) [reply mslice/models/workspacemanager/mantid_workspace_provider.py]: Seems doable. Should I open a separate issue?

Comment (martyngigg) [reply mslice/models/workspacemanager/mantid_workspace_provider.py]: I think that would be a good plan. You can never have too much test coverage!

**Comment by martyngigg** [comment]
@MatthewBowles I'm assuming this is ready to go now?

**Comment by MatthewBowles** [comment]
@martyngigg Yes

**Comment by martyngigg** [comment]
:+1:

**Pull request merged**